### DISCUSSION TITLE
libktx for Windows 7?

### DISCUSSION THREAD

**Discussion by samsonsite1** [KhronosGroup/KTX-Software#516] [Q&A] [answered]
Hello,
How to compile libktx under Windows 7?  I don't have Windows 10 or WSL (Windows Subsystem for Linux). 
bash.exe executable is obviously missing on my system, and it doesn't seem available either.

**Comment by MarkCallow** [comment, marked as answer]
Compiling requires C++11 (at least Visual Studio 2015 but VS 2017 or later is recommended as we have to disable a bunch of warnings in VS 2015) so the first thing to determine is if that version of VS or later can run on Windows 7. Only 64-bit builds are supported so you'll need a 64-bit Windows 7 to run the software. [Git for Windows](https://gitforwindows.org/) includes a `bash` executable that can be used. It is based on MinGW and is likely easier to install and more lightweight than Cygwin, the other obvious alternative for getting `bash`. 

Have you tried using the released Windows package on Windows 7? I do not have a Windows 7 machine so I can't try it myself.

**Reply by samsonsite1** [reply]
Git For Windows, that worked, thank you.

btw, compiling with VS 2015 fails for Release mode.

`2>  Generating Code...
2>c:\ktx-software-master\lib\astc-encoder\source\astcenc_decompress_symbolic.cpp(248): 
fatal error C1001: An internal error has occurred in the compiler.
2>  (compiler file 'f:\dd\vctools\compiler\utc\src\p2\main.c', line 255)
2>   To work around this problem, try simplifying or changing the program near the locations listed above.
2>  Please choose the Technical Support command on the Visual C++
2>   Help menu, or open the Technical Support help file for more information
`

To build astcenc-avx2-static under Release mode, I had to disable C/C++ Optimization for it.

**Reply by MarkCallow** [reply]
The VS 2015 Release builds are working in our CI. According to the CI purveyor, they have VS 2015 with Update 3. Are you running at least that version? A very similar problem  appeared in VS 2019 in one of its updates. A later update fixed it.
